### PR TITLE
Add validation

### PR DESCRIPTION
**Pull request by Tharuneshwarv**

### CODE REVIEW

File: backend/routes/uploadRoutes.js [hunk 1]
```diff
@@ -28,7 +30,13 @@ const fileFilter = (req, file, cb) => {
 
 const upload = multer({ storage, fileFilter }).single('image');
 
-router.post('/', upload, (req, res) => {
+const validator = {
+  upload: [
+    check('file').notEmpty().withMessage("Invalid file")
```
Comment (ajaybor0) [inline backend/routes/uploadRoutes.js:35]: Please fix the image upload validation, it's showing errors even when I upload a valid image.

File: backend/routes/paymentRoutes.js [hunk 2]
```diff
@@ -2,13 +2,36 @@ import express from 'express';
...
 import { protect } from '../middleware/authMiddleware.js';
 import { config, order, validate } from '../controllers/paymentController.js';
+import validateRequest from '../middleware/validator.js';
+import {body, check} from 'express-validator';
 
 const router = express.Router();
 
+const validator = {
+    order: [
+        check('body').exists().withMessage('Request body is required')
```
Comment (ajaybor0) [inline backend/routes/paymentRoutes.js:12]: Please fix the order validation, it checks for the existence of a property named 'body' in the request body. And there is no such property available in the request body.
```
        const options = req.body;
        console.log(options)  
//  {
//  amount: 8278850,
//    currency: "INR",
//   receipt: "receipt#65ad8fa293b9b1db8df33365"
// }
        
```

**Review by ajaybor0 — COMMENTED** [review]
Please see the comments above and update.

File: backend/routes/userRoutes.js [hunk 3]
```diff
@@ -15,24 +14,59 @@ import {
...
 import { protect, admin } from '../middleware/authMiddleware.js';
+import validateRequest from '../middleware/validator.js';
+import {body, param} from 'express-validator';
+
+const router = express.Router();
+const validator = {
+  checkLogin: [
+    body('email').trim().notEmpty().withMessage('Email is Required').bail().isEmail().withMessage("Please enter a valid email address"),
+    body('password').trim().isString().notEmpty().withMessage('Password is Empty')
+  ],
```
[the remark this answers is not in the document]
Comment (Tharuneshwarv) [reply backend/routes/userRoutes.js:25]: it is not a good practice to add length validation during login. We can add it during new user and reset-password

File: backend/routes/paymentRoutes.js (hunk 2 above)
Comment (Tharuneshwarv) [reply backend/routes/paymentRoutes.js]: fixed

File: backend/routes/uploadRoutes.js (hunk 1 above)
Comment (Tharuneshwarv) [reply backend/routes/uploadRoutes.js]: fixed

File: backend/routes/userRoutes.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ajaybor0) [reply backend/routes/userRoutes.js:25]: Agreed. Length validation is better suited for new user and reset-password processes.

**Review by ajaybor0 — APPROVED** [review]
@Tharuneshwarv Thanks for your contribution! Overall, the changes look good.

**Comment by ajaybor0** [comment]
🙏 Thanks for your collaboration!

**Pull request merged**